micro/test/service_test.go

```diff
@@ -1048,15 +1043,31 @@ func TestAddEndpoint_RaceCondition(t *testing.T) {
 	}
 	defer srv.Stop()
 
-	errs := make(chan error)
-	go func(errs chan error) {
-		errs <- srv.AddEndpoint("test", micro.ContextHandler(ctx, handler))
-	}(errs)
+	res := make(chan error, 10)
+	// now add a few endpoints concurrently
+	// and make sure they are added successfully
+	// and there is no race
+	for i := 0; i < 10; i++ {
+		go func(i int) {
+			res <- srv.AddEndpoint(fmt.Sprintf("test%d", i), micro.ContextHandler(ctx, handler))
```

Comment:
It could be me being picky, but I find it better in catching race conditions if you start adding endpoint exactly at the same time in goroutines. For example by having a wait group or channel that releases all goroutines when all gorutines have started?